examples/windows_nodepool_with_private_cluster/gke/main.tf

```diff
@@ -0,0 +1,95 @@
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "gke_private_cluster" {
+  source                     = "../modules/google_kubernetes_engine/gke_private_cluster/"
+  project_id                 = var.project
+  name                       = "test-gke-cluster"
+  region                     = var.region
+  network                    = var.network
+  subnetwork                 = var.subnetwork
+  ip_range_pods              = "pod-range"     #Note:pod-range is name of additional range in subnet. e.g : 10.2.0.0/16 or 10.2.4.0/21 (select wider range for pods as autoscalling will have large ip Requirements)
+  ip_range_services          = "service-range" #e.g : 10.2.4.0/24 
+  regional                   = var.regional
+  zones                      = var.zones
+  http_load_balancing        = true
+  network_policy             = true
+  horizontal_pod_autoscaling = true
+  enable_private_endpoint    = false
+  enable_private_nodes       = true
+  master_ipv4_cidr_block     = var.master_ipv4_cidr_block
+  remove_default_node_pool   = true
+  create_service_account     = false
+  service_account            = var.service_account
+  kubernetes_version         = var.gke_cluster_master_version
+  master_authorized_networks = var.master_authorized_networks
+  cluster_resource_labels    = var.gce_labels
+}
+
+#Linux Nodepool 
+module "gke_node_pool_01" {
+  source                         = "../modules/google_kubernetes_engine/gke_node_pool/"
```

Comment:
We should be passing these in via `node_pools= [...]`. Example https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/632#issuecomment-675912953
We can also hardcode as much as possible, like in the linked example.